test/TestRegulatorService.js

```diff
@@ -0,0 +1,91 @@
+const TestRegulatedToken = artifacts.require('./TestRegulatedToken.sol');
+const ServiceRegistry = artifacts.require('./ServiceRegistry.sol');
+const TestRegulatorService = artifacts.require('./TestRegulatorService.sol');
+
+const SUCCESS = 0;  /* 0  <= AMOUNT < 10 */
+const ELOCKED = 1;  /* 10 <= AMOUNT < 20 */
+const EDIVIS = 2;   /* 20 <= AMOUNT < 30 */
+const ESEND = 3;    /* 30 <= AMOUNT < 40 */
+const ERECV = 4;    /* 40 <= AMOUNT < 50 */
+
+contract('TestRegulatorService', async accounts => {
+  const owner = accounts[0];
+  const spender = owner;
+  const receiver = accounts[1];
+  const fromOwner = { from: owner };
+  const fromReceiver = { from: receiver };
+
+  let regulator;
+  let token;
+
+  beforeEach(async () => {
+    regulator = await TestRegulatorService.new({ from: owner });
+    const registry = await ServiceRegistry.new(regulator.address);
+    token = await TestRegulatedToken.new(registry.address, 'Test', 'TEST');
+  });
+
+  const assertBalances = async balances => {
```

Comment:
Can you create a helper for this so we aren't duplicating code?

Reply:
I agree that assertBalances() and checkResult() are annoying duplicated code. A lot of their value comes from having closures on local variables. The only ways I can think of to reduce boiler plate are to pass in more parameters to the functions or to create a Factory method that returns an instance of assertBalances() with the needed closures. Both those options seem like more trouble than they are worth. I think we are best off leaving for now and refactoring if we see ourselves doing this more.

I'm open to suggestions.